src/tuntap/tun.go

```diff
@@ -0,0 +1,361 @@
+package tuntap
```

Comment:
I still don't like that this file is "new" instead of recognized as a move, but I guess there's no point in holding things up longer just to try and fix that.

Nothing looks obviously wrong, but since it's not a diff, I can't guarantee that I noticed any nontrivial changes.